### ISSUE TITLE
I can't execute a javascript inside the HTML script tag

### ISSUE THREAD

**Issue by emilioriosvz** [casperjs/casperjs#1326]
I'm currently testing a site which attempts an automatic redirection (callback url) when the user tries to navigate to some pages. The replied by the server "callback url" HTML looks like this:

```
<!DOCTYPE html><html lang="en">
<head>
<title>Huballin |&nbsp;Callbackurl</title>
</head>
<body>
    <script>
        (function(){
            console.log('hello world')
                        // Few logics which generates a next named variable (url-like)
            document.location.href = next;
        })();
    </script>

</body></html>
```

So, after this quick execution it always redirects to the homepage or the -- requested by the user -- page. That's fine, works nice in local, qa and production environments.

However, for casperjs seems impossible to execute it. After going to "callback url" it totally stops. I've checked the `getHTML()` outputs and it doesn't go beyond the "callback url" one.

I've checked another navigations; those without the "callback url" requirement. And also, I've tested other Javascript snipets. And all of these tests worked like a charm. 

In my humble and unexperienced observation, the only thing that I think changes is the `script` tag. The other javascript pieces are requested in the classical way.  

This is the code of my test:

```
casper.test.begin('Sign in test', function(test) {
   casper.start('http://localhost:8100/#/signin');  // I've also tried by open the homepage and the clicking here

   casper.then(function(){
    this.echo(this.getHTML())
   })
   casper.run(function() {
    test.done();
   });
});
```

I've also tried to set up some `pageSettings`:

```
casper.options.loadPlugins = true
casper.options.userAgent = 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1588.0 Safari/537.36'
casper.options.javascriptEnabled = true
casper.options.localToRemoteUrlAccessEnabled = true
```

**tl;dr**
Why this snippet doesn't run?

Thanks in advance,
any log, extra info, or setting that you need is available I didn't added anything more because I'm not sure nor what's happening nor how does casperjs work internally.

**Comment by amerikan** [comment]
So it seems like when your page loads, it then redirects to another page and it is there where you are try to echo out some html?

When loading a page, you usually want to add a `waitForSelector` or one of the many other **waits** available in casperjs before running a certain script. For example, if the next page contains some `<div id="foo">` then you want to add something like:

```js
casper.waitForSelector('#foo', function () {
    // do whatever after that section
    this.echo(this.getHTML())
});
```
This way you will force casperjs to wait for the `<div id="foo">` to exist before continuing.

**Comment by kensoh** [comment]
Hi @emilioriosvz I've been a CasperJS user for past 2 years without encountering this issue. Not sure if this has been fixed or is it still happening for you?

**Comment by kensoh** [comment]
Thanks @amerikan for sharing your inputs!

Hi @emilioriosvz closing this issue for now as it isn't really replicable with a locally hosted website, and it is likely to be fixed by now a fundamental feature of running JS scripts on the webpage.

But do reopen if you are still experiencing it on latest CasperJS v1.1.4 and PhantomJS v2.1.1. Thanks for raising this!
